fusion-plugin-font-loader-react/src/plugin.js

```diff
@@ -26,26 +23,38 @@ import type {
   StyledFontsObjectType,
 } from './types.js';
 
+let preloadSession;
+
 const plugin = createPlugin({
   deps: {
     config: ConfigToken,
   },
+  provides: ({config}) => {
+    if (!preloadSession) {
+      const {fonts, preloadDepth} = config;
+      const atomicFonts: AtomicFontsObjectType = (fonts: any);
+      const fallbackLookup = generateFallbackMap(
+        atomicFonts,
+        preloadDepth || 0
+      );
+      preloadSession = new PreloadSession(fallbackLookup);
+    }
+    return preloadSession.getFontDetails;
+  },
   middleware: ({config}) => {
     const {fonts, preloadDepth, withStyleOverloads, preloadOverrides} = config;
     const atomicFonts: AtomicFontsObjectType = (fonts: any);
     const styledFonts: StyledFontsObjectType = (fonts: any);
-    const fallbackLookup = generateFallbackMap(atomicFonts, preloadDepth || 0);
-    const preloadSession = new PreloadSession(fallbackLookup);
+    if (!preloadSession) {
```

Comment:
Can you get away with just pulling `getFontDetails` from the second parameter to `middleware`?

Reply:
Sorry just seen this comment, not quite understanding. I don't think I use `getFontDetails` in `middleware`

Reply:
Oh sorry, not `getFontDetails`, I meant `preloadSession`. I was just curious why you need to create the global `preloadSession` in both `provides` and `middleware` since it's the same logic. As far as I know, the `provides` logic will be run at startup, before we handle any requests with middleware.

Reply:
That's what I thought too, but  when I tested there seems to be a race condition. Should provides always be the second argument to middleware?

Reply:
Thats weird. It seems like you should just be able to access `preloadSession` globally once you're in the middleware.